src/accountsdb/db.zig

```diff
@@ -1645,15 +1692,15 @@ pub const AccountsDB = struct {
 
         var total_accounts_deleted: u64 = 0;
         for (shrink_account_files) |shrink_file_id| {
-            var shrink_account_file_rw = blk: {
-                const file_map, var file_map_lg = self.file_map.readWithLock();
+            self.file_map_safe.fd_keep_open_rw.lock();
```

Comment:
this operation is what im worried about - holding this lock for too long ....

talked offline: should move this to queue the shrink_account_file into delete_account files

Reply:
f175b944bceaefa3099b363f7b097a71d1f392f1